api/api_test.go

```diff
@@ -114,6 +126,21 @@ func TestFailedToAddFilterBlueprint(t *testing.T) {
 		So(response, ShouldResemble, "Instance not found\n")
 	})
 
+	Convey("When instance is unpublished and the request is not authenticated, a not found error is returned", t, func() {
+		reader := strings.NewReader(`{"instance_id":"12345678"}`)
+		r, err := http.NewRequest("POST", "http://localhost:22100/filters", reader)
+		So(err, ShouldBeNil)
+
+		w := httptest.NewRecorder()
+		api := routes(authHeader, host, mux.NewRouter(), &mocks.DataStore{}, &mocks.FilterJob{}, &mocks.DatasetAPI{Unpublished: true}, previewMock)
+		api.router.ServeHTTP(w, r)
+		So(w.Code, ShouldEqual, http.StatusBadRequest)
```

Comment:
Mismatch in title of test and assertion (e.g. the expected response status), title refers to a not found error but the actual response status in assertion is bad request?

The dataset API will be responding with not found in the near future (PR about to go out)